x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java

```diff
@@ -250,19 +255,35 @@ public void testSearchableSnapshotShardsAreSkippedBySearchRequestWithoutQuerying
 
         final IndexMetadata updatedIndexMetadata = getIndexMetadata(searchableSnapshotIndexOutsideSearchRange);
         final IndexLongFieldRange updatedTimestampMillisRange = updatedIndexMetadata.getTimestampRange();
-        final DateFieldMapper.DateFieldType dateFieldType = indicesService.getTimestampFieldType(updatedIndexMetadata.getIndex());
+        final IndexLongFieldRange updatedEventIngestedMillisRange = updatedIndexMetadata.getEventIngestedRange();
+
+        fieldTypeMap = indicesService.getTimestampFieldTypeMap(updatedIndexMetadata.getIndex());
+        final DateFieldMapper.DateFieldType dateFieldType = fieldTypeMap.get(DataStream.TIMESTAMP_FIELD_NAME);
         assertThat(dateFieldType, notNullValue());
         final DateFieldMapper.Resolution resolution = dateFieldType.resolution();
         assertThat(updatedTimestampMillisRange.isComplete(), equalTo(true));
+        assertThat(updatedEventIngestedMillisRange.isComplete(), equalTo(true)); /// MP TODO: this sometimes fails - why?
```

Comment:
The assert on `updatedTimestampMillisRange.isComplete()` never fails.

Out of ~30 runs I've done of this test, the `updatedEventIngestedMillisRange.isComplete()` fails roughly 80% of the time. I tried to find any correlations of what it fails with (such as the several randomized factors in this test), but was unable to find any correlation. I also tried adding a 10s sleep to see if there was some race condition that need time, but it fails sporadically even with the sleep.

So somewhere there is proper handling of the `@timestamp` IndexLongFieldRange to provide info on all shards that is not happening for the newly added `event.ingested` range, but I'm not able to figure out what/where that is.